aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/continuations/ChallengeContinuation.java

```diff
@@ -140,6 +160,9 @@ public void continueTask() {
         respondToAuthChallengeRequest.setSession(challengeResult.getSession());
         respondToAuthChallengeRequest.setClientId(clientId);
         respondToAuthChallengeRequest.setChallengeResponses(challengeResponses);
+        if (clientMetaData != null) {
```

Comment:
Would the operation that executes `respondToAuthChallengeRequest` have any pre conditions such as clientMetadata cannot be null or an empty map?

What happens when the map is empty?

Reply:
ClientMetadata can be null when passed to `respondToAuthChallengeRequest ` but will not be passed to the service.
If clientMetadata is empty, it is passed down as an empty map to the lambda triggers. Verified that empty map is appropriately marshalled by the `RespondToAuthChallengeRequest` marshaller.